packages/typegpu-wgsl-parser/src/grammar.ne

```diff
@@ -149,15 +149,22 @@ template_arg_comma_list ->
   expression ("," expression):* ",":? {% ([first, rest]) => [first, ...rest.map(tuple => tuple[1])] %}
 
 @{%
+export type FunctionArgument = { type: 'func_argument', ident: string, typespec: TypeSpecifier };
 export type FunctionDecl = { type: 'function_decl', header: FunctionHeader, body: CompoundStatement, attrs: Attribute[] };
-export type FunctionHeader = { type: 'function_header', identifier: string };
+export type FunctionHeader = { type: 'function_header', ident: string, returntype: ReturnType | null, args: FunctionArgument[] | null };
+export type ReturnType = { type: 'return_type', typespec: TypeSpecifier };
 
 %}
+return_type ->
+  "-" ">" type_specifier {% ([,, typespec]) => ({ type: 'return_type', typespec }) %}
```

Comment:
I think this would accept whitespace between `-` and `>`, which is not permitted in WGSL. Lets add a new token `->` and place it before the minus sign, so that it gets recognized first.

```suggestion
  "->" type_specifier {% ([, typespec]) => ({ type: 'return_type', typespec }) %}
```